TVC_MVP/CountyPresenter.swift

```diff
@@ -81,27 +71,32 @@ class CountyPresenter: CountyPresnterProtocol {
                         guard let id = county[Constants.id] as? Int else { return }
                         let countyObject = County(countyID: id, countyName: name)
                         countyArray.append(countyObject)
-                        reloadData()
                     }
                 } else {
-                    presentParseErrorAlert()
+                    presentJSONErrorAlert()
                 }
             } else {
-                presentParseErrorAlert()
+                presentJSONErrorAlert()
             }
         } catch {
-            presentParseErrorAlert()
             print(error.localizedDescription)
         }
+        reloadData()
```

Comment:
It's not bad, but now we would be reloading the data even if there is nothing to reload, since there could have been an error while parsing. The reload should happen after the loop.